doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_ironside.rst

```diff
@@ -75,6 +225,124 @@ The following UICR fields are supported:
    If no UICR values are programmed, |ISE| applies a set of default configurations.
    Applications that do not require custom settings can rely on these defaults without modifying the UICR.
 
+UICR image generation
+=====================
+
+The UICR image is an automatically created build artifact that generates the UICR content and peripheral configuration (PERIPHCONF) entries.
+
+What is the UICR image
+----------------------
+
+The UICR image consists of several components:
+
+* UICR configuration - The main UICR structure containing version information, security settings, memory protection, and boot configuration
+* PERIPHCONF entries - Peripheral configuration data extracted from the devicetree that defines peripheral access permissions and pin multiplexing
+
+The UICR image is generated as a HEX file.
+The UICR HEX file is automatically programmed when using ``west flash`` and is located at ``build/uicr/zephyr/zephyr.hex`` relative to your project's build directory.
+
+UICR image generation process
+-----------------------------
+
+The following steps process describe the UICR image generation process:
+
+1. Devicetree analysis: The build system analyzes all devicetrees in the sysbuild build to identify peripheral usage and pin assignments across all images.
+2. PERIPHCONF generation: Python scripts generate C source code containing UICR configuration macros based on the devicetree.
+3. Compilation: The generated C code is compiled into ELF binaries containing PERIPHCONF sections.
+4. ELF extraction: The UICR generator extracts PERIPHCONF data from ELF files and merges them together.
```

Comment:
Merges them together where? It is placed in the periphconf hex file?

Reply:
Now the section above is explicitly documenting a periphconf.hex file, so maybe this is clear now?